pkg/controller/migplan/pvlist.go

```diff
@@ -0,0 +1,117 @@
+package migplan
+
+import (
+	"context"
+	migapi "github.com/fusor/mig-controller/pkg/apis/migration/v1alpha1"
+	migref "github.com/fusor/mig-controller/pkg/reference"
+	core "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type PvMap map[types.NamespacedName]core.PersistentVolume
+type Claims []types.NamespacedName
+
+// Update the PVs listed on the plan.
+func (r *ReconcileMigPlan) updatePvs(plan *migapi.MigPlan) error {
+	planResources, err := plan.GetRefResources(r)
+	if err != nil {
+		return err
+	}
+	client, err := planResources.SrcMigCluster.GetClient(r)
+	if err != nil {
+		return err
+	}
+
+	plan.Spec.BeginPvStaging()
+
+	// Build PV map.
+	table, err := r.getPvMap(client)
+	if err != nil {
+		return err
+	}
+	namespaces := planResources.MigAssets.Spec.Namespaces
+	claims, err := r.getClaims(client, namespaces)
+	if err != nil {
+		return err
+	}
+	for _, claim := range claims {
+		pv, found := table[claim]
+		if !found {
+			continue
+		}
+		plan.Spec.AddPv(
+			migapi.PersistentVolume{
+				Name:         pv.Name,
+				StorageClass: pv.Spec.StorageClassName,
+			})
+	}
+
+	// Set the condition to indicate that discovery has been performed.
+	plan.Status.SetCondition(migapi.Condition{
+		Type:     PvsDiscovered,
+		Status:   True,
+		Reason:   Done,
+		Category: migapi.Required,
+		Message:  PvsDiscoveredMesssage,
+	})
+
+	// Update
+	plan.Spec.PersistentVolumes.EndPvStaging()
+	err = r.Update(context.TODO(), plan)
+	if err != nil {
+		return nil
+	}
+
+	return nil
+}
+
+// Get a table (map) of PVs keyed by namespaced name.
+func (r *ReconcileMigPlan) getPvMap(client k8sclient.Client) (PvMap, error) {
+	table := PvMap{}
+	list := core.PersistentVolumeList{}
+	err := client.List(context.TODO(), &k8sclient.ListOptions{}, &list)
+	if err != nil {
+		return nil, err
+	}
+	for _, pv := range list.Items {
+		claim := pv.Spec.ClaimRef
+		if migref.RefSet(claim) {
+			key := types.NamespacedName{
+				Namespace: claim.Namespace,
+				Name:      claim.Name,
+			}
+			table[key] = pv
+		}
+	}
+
+	return table, nil
+}
+
+// Get a list of PVCs found on pods with the specified namespaces.
+func (r *ReconcileMigPlan) getClaims(client k8sclient.Client, namespaces []string) (Claims, error) {
```

Comment:
So to get the `PersistentVolumeClaims`, we have to look at all of the pods in a namespace. Good to see this written out, helped me understand.